test/functional/importer_controller_test.rb

```diff
@@ -20,64 +22,164 @@ def setup
     @issue.reload
     assert_equal 'barfooz', @issue.subject
   end
 
+  test 'should send email when Send email notifications checkbox is checked and issue added' do
+    assert_equal 'foobar', @issue.subject
+    Mailer.expects(:deliver_issue_add)
+
+    assert_equal 0, Issue.where(subject: 'barfooz').count
+    post :result, params: build_params(send_emails: 'true')
+    assert_response :success
+    assert_equal 1, Issue.where(subject: 'barfooz').count
+  end
+
   test 'should NOT send email when Send email notifications checkbox is unchecked' do
     assert_equal 'foobar', @issue.subject
     Mailer.expects(:deliver_issue_edit).never
 
-    post :result, build_params(:send_emails => nil)
+    post :result, params: build_params(update_issue: 'true')
     assert_response :success
     @issue.reload
     assert_equal 'barfooz', @issue.subject
   end
 
   test 'should add watchers' do
-    assert issue_has_none_of_these_watchers?(@issue, [@user, @user.parent])
-    post :result, build_params
+    assert issue_has_none_of_these_watchers?(@issue, [@user])
+    post :result, params: build_params(update_issue: 'true')
+    assert_response :success
+    @issue.reload
+    assert issue_has_all_of_these_watchers?(@issue, [@user])
+  end
+
+  test 'should handle key value list value' do
+    Mailer.expects(:deliver_issue_add).never
+    IssueCustomField.where(name: 'Area').each { |icf| icf.update(multiple: false) }
+    @iip.destroy
+    @iip = create_iip!('KeyValueList', @user, @project)
+    post :result, params: build_params
+    assert_response :success
+    assert keyval_vals_for(Issue.find_by!(subject: 'パンケーキ')) == ['Tokyo']
+    assert keyval_vals_for(Issue.find_by!(subject: 'たこ焼き')) == ['Osaka']
+    assert Issue.find_by(subject: 'サーターアンダギー').nil?
+  end
+
+  test 'should handle multiple key value list values' do
+    Mailer.expects(:deliver_issue_add).never
+    @iip.destroy
+    @iip = create_iip!('KeyValueListMultiple', @user, @project)
+    post :result, params: build_params
+    assert_response :success
+    assert keyval_vals_for(Issue.find_by!(subject: 'パンケーキ')) == ['Tokyo']
+    assert keyval_vals_for(Issue.find_by!(subject: 'たこ焼き')) == ['Osaka']
+    issue = Issue.find_by!(subject: 'タピオカ')
+    assert ['Tokyo', 'Osaka'].all? { |area| area.in?(keyval_vals_for(Issue.find_by!(subject: 'タピオカ'))) }
+    assert Issue.find_by(subject: 'サーターアンダギー').nil?
+  end
+
+  test 'should error when assigned_to is missing' do
+    @iip.update!(csv_data: "#,Subject,assigned_to\n#{@issue.id},barfooz,JohnDoe\n")
+    @issue.update!(assigned_to: @user)
+    post :result, params: build_params(update_issue: 'true').tap { |params| params[:fields_map]['assigned_to'] = 'assigned_to' }
+    assert_response :success
+    assert response.body.include?('Warning')
+    @issue.reload
+    assert_equal 'foobar', @issue.subject
+    assert_equal @user, @issue.assigned_to
+  end
+
+  test 'should error when assigned_to user is not assignable' do
```

Comment:
Will users see an error in this case? Maybe did you mean like this?

```suggestion
  test 'should unset assigned_to when assigned_to user is not assignable' do
```

Reply:
That is correct, thank you 🙏